### ISSUE TITLE
feat: export/import app data

### ISSUE THREAD

**Issue by castdrian** [khcrysalis/Feather#183]
since the author of https://github.com/khcrysalis/Feather/issues/14 closed the issue, I've actually got a good use case for this:

I debug my Theos tweaks on jailed environments frequently, and on more than one occasion accidentally wiped the user defaults with no easy way to restore, so being able to just reinstall with backed up defaults would probably make the lives of tweak devs a lot easier

**Comment by sevenpastzeero** [comment]
This would be useful. Since sideloaded app cannot connect to iCloud (even paid certs with icloud support), we can export app data from one device to another.

**Comment by binnichtaktiv** [comment]
Maybe try [my tweak](https://t.me/binnichtaktivsipas/95044). Its not perfect but it worked for the apps I use

**Comment by khcrysalis** [comment]
is there source code to that @binnichtaktiv

**Comment by binnichtaktiv** [comment]
> is there source code to that [@binnichtaktiv](https://github.com/binnichtaktiv)

Can I send it to you somewhere else?

**Comment by khcrysalis** [comment]
No. But is there a github repository for it that contains its source code?

**Comment by binnichtaktiv** [comment]
> No. But is there a github repository for it that contains its source code?

Yes but its still private. I invited you I hope thats okay

**Comment by khcrysalis** [comment]
Well I would've preferred it to be public, incase of potential malicious stuff

**Comment by binnichtaktiv** [comment]
> Well I would've preferred it to be public, incase of potential malicious stuff

Its just ugly and trash code thats why its private, but if you only want to test it if its public then I can set it to public

**Comment by binnichtaktiv** [comment]
its public now but pls dont laught the code is bad😂
https://github.com/binnichtaktiv/AppDataBackup
@khcrysalis

**Comment by sevenpastzeero** [comment]
> Maybe try [my tweak](https://t.me/binnichtaktivsipas/95044). Its not perfect but it worked for the apps I use

I used your tweak before, great tweak saved me a hassle with a few games. But unfortunately it does not work with everything. Sometimes the app just crash (Apollo for example).

**Comment by castdrian** [comment]
The approach is the same as a certain sideloading app that hands out stolen signing assets, it’s nice to see that someone wrote an open source version of that

**Comment by sevenpastzeero** [comment]
> The approach is the same as a certain sideloading app that hands out stolen signing assets, it’s nice to see that someone wrote an open source version of that

How can we determine if a modified ipa, like YT, or a social app, is safe from this stealing sideloading issue? Is there a specific framework to look for? Or we just trust in the community hoping they are sharing trustworthy ipas?

**Comment by binnichtaktiv** [comment]
> > Maybe try [my tweak](https://t.me/binnichtaktivsipas/95044). Its not perfect but it worked for the apps I use
> 
> I used your tweak before, great tweak saved me a hassle with a few games. But unfortunately it does not work with everything. Sometimes the app just crash (Apollo for example).

I have to check it. Is it only apollo?

**Comment by binnichtaktiv** [comment]
> > The approach is the same as a certain sideloading app that hands out stolen signing assets, it’s nice to see that someone wrote an open source version of that
> 
> How can we determine if a modified ipa, like YT, or a social app, is safe from this stealing sideloading issue? Is there a specific framework to look for? Or we just trust in the community hoping they are sharing trustworthy ipas?

What do you mean? I think castdrian meant free certs that get leaked.

**Comment by sevenpastzeero** [comment]
> I have to check it. Is it only apollo?

Apollo (wanted to easily migrate my stats and and settings), Noteshelf3, ChatGPT (I wanted to copy my chats without signing up).


> What do you mean? I think castdrian meant free certs that get leaked.

Oh I thought that it was like a hidden dylib that export the app data.

**Comment by bodesamir** [comment]
> its public now but pls dont laught the code is bad😂 https://github.com/binnichtaktiv/AppDataBackup [@khcrysalis](https://github.com/khcrysalis)

not working for feather 2.0 it was working for old versions of feather app
